Shared/Samples/Add custom dynamic entity data source/AddCustomDynamicEntityDataSourceView.swift

```diff
@@ -0,0 +1,164 @@
+// Copyright 2023 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+
+struct AddCustomDynamicEntityDataSourceView: View {
+    /// A map with an ArcGIS oceans basemap style.
+    @State var map: Map = {
+        let map = Map(basemapStyle: .arcGISOceans)
+        map.initialViewpoint = Viewpoint(
+            latitude: 47.984,
+            longitude: -123.657,
+            scale: 3e6
+        )
+        return map
+    }()
+    
+    /// The dynamic entity layer that is displaying our custom data.
+    @State var dynamicEntityLayer: DynamicEntityLayer
+    
+    /// The point on the screen the user tapped.
+    @State private var tappedScreenPoint: CGPoint?
+    
+    /// The placement of the callout.
+    @State private var calloutPlacement: CalloutPlacement?
+    
+    init() {
```

Comment:
Extract the logic of creating a custom dynamic entity layer from the initializer to a separate static method (`private static func makeDynamicEntityLayer() -> DynamicEntityLayer`), and/or add step by step comments, may be beneficial for a user.

Reply:
Yes, doing things in the initializer of the view is definitely not the way to go…using a model would almost be preferred. Not that I'm suggesting that, but the way it is now, if the parent's body were to be re-computed, then this initializer would be called and each of these objects would be re-created. The map should be created in a function (`makeMap()`) that is called to initialize the `map` property and then the layer could just be retrieved from the operational layers using a computed property. Or, if you really want to, go back to using a model.

Reply:
> Not that I'm suggesting that, but the way it is now, if the parent's body were to be re-computed, then this initializer would be called and each of these objects would be re-created. 

Good point.